src/main/java/com/google/api/codegen/GapicContext.java

```diff
@@ -36,42 +36,49 @@
 public class GapicContext extends CodegenContext {
 
   private final Model model;
-  private final ApiConfig apiConfig;
+  private final GapicProductConfig productConfig;
 
   private final ServiceMessages serviceMessages;
-  private final ServiceConfig serviceConfig;
+  private final ProductServiceConfig productServiceConfig;
 
   public static final String API_WRAPPER_SUFFIX = "Client";
 
   /** Constructs the abstract instance. */
-  protected GapicContext(Model model, ApiConfig apiConfig) {
+  protected GapicContext(Model model, GapicProductConfig productConfig) {
     this.model = Preconditions.checkNotNull(model);
-    this.apiConfig = Preconditions.checkNotNull(apiConfig);
+    this.productConfig = Preconditions.checkNotNull(productConfig);
     this.serviceMessages = new ServiceMessages();
-    this.serviceConfig = new ServiceConfig();
+    this.productServiceConfig = new ProductServiceConfig();
   }
 
   /** Returns the associated model. */
   public Model getModel() {
     return model;
   }
 
-  /** Returns the associated config. */
-  public ApiConfig getApiConfig() {
-    return apiConfig;
+  /*
+   * Returns the associated config.
+   *
+   * NOTE: The name here is out of date, but this whole class will be deprecated soon.
```

Comment:
Well it would be a lot of disruption for little gain - I didn't want to touch templates.

This is the big advantage of the MVVM architecture - changes to the transformers don't affect the templates.